src/main/java/duke/command/CommandDone.java

```diff
@@ -0,0 +1,41 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.ui.Ui;
+
+import java.io.IOException;
+
+/**
+ * The class that models a mark as done command.
+ */
+public class CommandDone extends Command {
+    int index;
+
+    /**
+     * Instantiates an object with given index to mark the task as done when
+     * called to execute.
+     * @param index The index where the task should be deleted.
+     */
+    public CommandDone(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Execute the mark as done action and auto-saves to the file when called.
```

Comment:
For header comments, should leave a line in between the Description of the method and the description of the params.